src/code/m_lib.c

```diff
@@ -0,0 +1,763 @@
+            if (diff >= 0) {
+                *pValue += minStep;
+                if ((s16)(target - *pValue) < 0) {
+                    *pValue = target;
+                }
+            } else {
+                *pValue -= minStep;
+                if ((s16)(target - *pValue) > 0) {
+                    *pValue = target;
+                }
+            }
+        }
+    }
+    return target - *pValue;
+}
+
+s16 add_calc_short_angle3(s16* pValue, s16 target, f32 fraction, s16 maxStep, s16 minStep) {
+    f32 stepSize;
+    s32 uTarget;
+    s32 newValue;
+    s32 uValue;
+
+    if (target != *pValue) {
+        uValue = (u16)*pValue;
+        uTarget = (u16)target;
+
+        if (uValue > uTarget) {
+            uTarget += 0x10000;
+        }
+
+        stepSize = (uTarget - uValue) * fraction;
+
+        if (stepSize > maxStep) {
+            stepSize = maxStep;
+        } else if (stepSize < minStep) {
+            stepSize = minStep;
+        }
+
+        newValue = uValue + (s32)stepSize;
+        if (newValue > uTarget) {
+            newValue = uTarget;
+        }
+        *pValue = newValue;
+    }
+
+    return target - *pValue;
+}
+
+void adds(s16* pValue, s16 target, s16 scale, s16 maxStep) {
+    s16 diff = target - *pValue;
+    diff /= scale;
+
+    if (diff > maxStep) {
+        *pValue += maxStep;
+        return;
+    }
+
+    if (diff < -maxStep) {
+        *pValue -= maxStep;
+        return;
+    }
+
+    *pValue += diff;
+}
+
+void rgba_t_move(Color_RGBA8* dest, Color_RGBA8* src) {
+    dest->r = src->r;
+    dest->g = src->g;
+    dest->b = src->b;
+    dest->a = src->a;
+}
+
+s32 none_proc1() {
+    return 0;
+}
+
+s32 none_proc2(Actor* actor UNUSED, Game_Play* play UNUSED) {
+}
+
+void Cheap_gfx_display(Game_Play* play, Gfx* dl) {
+
+    if ((!(&play->state)) && (!(&play->state))) {}
+
+    OPEN_DISPS(play->state.gfxCtx);
+
+    _texture_z_light_fog_prim(play->state.gfxCtx);
+
+    gSPMatrix(POLY_OPA_DISP++, _Matrix_to_Mtx_new(play->state.gfxCtx), G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+
+    gSPDisplayList(POLY_OPA_DISP++, dl);
+
+    CLOSE_DISPS(play->state.gfxCtx);
+}
+
+void Cheap_gfx_display_xlu(Game_Play* play, Gfx* dl) {
+
+    if ((!(&play->state)) && (!(&play->state))) {}
+
+    OPEN_DISPS(play->state.gfxCtx);
+
+    _texture_z_light_fog_prim_xlu(play->state.gfxCtx);
+
+    gSPMatrix(POLY_XLU_DISP++, _Matrix_to_Mtx_new(play->state.gfxCtx), G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+
+    gSPDisplayList(POLY_XLU_DISP++, dl);
+
+    CLOSE_DISPS(play->state.gfxCtx);
+}
+
+s32 _Game_play_isPause(Game_Play* play) {
+
+    return play->pause.enabled != 0;
+}
+
+void* Lib_SegmentedToVirtual(void* ptr) {
+    return (void*)SEGMENTED_TO_VIRTUAL(ptr);
+}
+
+void* Lib_SegmentedToVirtualNull(void* ptr) {
+    if (((uintptr_t)ptr >> 28) == 0) {
+        return ptr;
+    } else {
+        return (void*)SEGMENTED_TO_VIRTUAL(ptr);
+    }
+}
+
+void* PhysicalToVirtual(uintptr_t ptr) {
+
+    if (ptr == 0) {
```

Comment:
Could you remove this kind of empty line at the beginning of functions?
```suggestion
void* PhysicalToVirtual(uintptr_t ptr) {
    if (ptr == 0) {
```